MaterialDesignThemes.Wpf/Themes/MaterialDesignTheme.ScrollViewer.xaml

```diff
@@ -10,15 +12,96 @@
-            <ScrollBar x:Name="PART_HorizontalScrollBar" AutomationProperties.AutomationId="HorizontalScrollBar" Cursor="Arrow" Grid.Column="0" Maximum="{TemplateBinding ScrollableWidth}" Minimum="0" Orientation="Horizontal" Grid.Row="1" Visibility="{TemplateBinding ComputedHorizontalScrollBarVisibility}" Value="{Binding HorizontalOffset, Mode=OneWay, RelativeSource={RelativeSource TemplatedParent}}" ViewportSize="{TemplateBinding ViewportWidth}" />
+
+            <Rectangle 
+                x:Name="PART_Corner" 
+                Grid.Column="1" 
+                Fill="{DynamicResource MaterialDesignSelection}" 
+                Grid.Row="1"/>
+
+            <ScrollContentPresenter 
+                x:Name="PART_ScrollContentPresenter" 
+                CanContentScroll="{TemplateBinding CanContentScroll}"
+                CanHorizontallyScroll="False" CanVerticallyScroll="False"
+                ContentTemplate="{TemplateBinding ContentTemplate}"
+                Content="{TemplateBinding Content}"
+                Grid.Column="0" 
+                Margin="{TemplateBinding Padding}"
+                Grid.Row="0"/>
+
+            <ScrollBar 
+                x:Name="PART_VerticalScrollBar" 
+                AutomationProperties.AutomationId="VerticalScrollBar"
+                Cursor="Arrow" 
+                Grid.Column="1"
+                Maximum="{TemplateBinding ScrollableHeight}"
+                Minimum="0" 
+                Grid.Row="0" 
+                Visibility="{TemplateBinding ComputedVerticalScrollBarVisibility}" 
+                Value="{Binding VerticalOffset, Mode=OneWay, RelativeSource={RelativeSource TemplatedParent}}" 
+                ViewportSize="{TemplateBinding ViewportHeight}" />
+
+            <ScrollBar 
+                x:Name="PART_HorizontalScrollBar"
+                AutomationProperties.AutomationId="HorizontalScrollBar" 
+                Cursor="Arrow" 
+                Grid.Column="0" 
+                Maximum="{TemplateBinding ScrollableWidth}" 
+                Minimum="0" 
+                Orientation="Horizontal" 
+                Grid.Row="1" 
+                Visibility="{TemplateBinding ComputedHorizontalScrollBarVisibility}"
+                Value="{Binding HorizontalOffset, Mode=OneWay, RelativeSource={RelativeSource TemplatedParent}}" 
+                ViewportSize="{TemplateBinding ViewportWidth}" />
         </Grid>
+        <ControlTemplate.Triggers>
+            <Trigger Property="wpf:ScrollViewerAssist.CornerRectangleVisibility" Value="Collapsed">
+                <Setter TargetName="PART_Corner" Property="Visibility" Value="Collapsed"/>
+            </Trigger>
+
+            <Trigger Property="wpf:ScrollViewerAssist.CornerRectangleVisibility" Value="Hidden">
+                <Setter TargetName="PART_Corner" Property="Visibility" Value="Hidden"/>
+            </Trigger>
```

Comment:
Could these be moved into a TemplateBinding on the `PART_Corner`?

Reply:
Yes it's better option for setting CornerRectangleVisibility.